cpp/cmake/thirdparty/get_arrow.cmake

```diff
@@ -126,9 +126,14 @@ function(find_and_configure_arrow VERSION BUILD_STATIC ENABLE_S3 ENABLE_ORC ENAB
     # Turn off CPM using `find_package` so we always download and make sure we get proper static
     # library.
     set(CPM_DOWNLOAD_Arrow TRUE)
+    # We don't want to explicitly force a static OpenSSL when `BUILD_STATIC` so we allow the
```

Comment:
This feels like it’s phrased a little backwards. Maybe explain the default behavior and how users can change it. The comment below this one follows that approach and makes more sense for a first-time reader in my opinion. For example:

By default ARROW will try to search for a static version of OpenSSL which is a bad idea given that shared linking is advised for critical components like SSL. If a static build is requested, we honor ARROW's default of static linking, but users may consider setting `ARROW_OPENSSL_USE_SHARED` even in static builds.